multimodal/src/autogluon/multimodal/data/process_ovd.py

```diff
@@ -0,0 +1,200 @@
+            Whether to require feature column information in dataloader.
+        """
+
+        self.prefix = model.prefix
+        self.missing_value_strategy = missing_value_strategy
+        self.requires_column_info = requires_column_info
+
+        self.max_img_num_per_col = max_img_num_per_col
+        if max_img_num_per_col <= 0:
+            logger.debug(f"max_img_num_per_col {max_img_num_per_col} is reset to 1")
+            max_img_num_per_col = 1
+        self.max_img_num_per_col = max_img_num_per_col
+        logger.debug(f"max_img_num_per_col: {max_img_num_per_col}")
+
+    @property
+    def image_key(self):
+        return f"{self.prefix}_{IMAGE}"
+
+    @property
+    def image_meta_key(self):
+        return f"{self.prefix}_{IMAGE_META}"
+
+    @property
+    def prompt_key(self):
+        return f"{self.prefix}_{PROMPT}"
+
+    @property
+    def image_valid_num_key(self):
+        return f"{self.prefix}_{IMAGE_VALID_NUM}"
+
+    @property
+    def image_column_prefix(self):
+        return f"{self.image_key}_{COLUMN}"
+
+    def collate_fn(self, image_column_names: Optional[List] = None, per_gpu_batch_size: Optional[int] = None) -> Dict:
+        """
+        Collate images into a batch. Here it pads images since the image number may
+        vary from sample to sample. Samples with less images will be padded zeros.
+        The valid image numbers of samples will be stacked into a vector.
+        This function will be used when creating Pytorch DataLoader.
+
+        Returns
+        -------
+        A dictionary containing one model's collator function for image data.
+        """
+        fn = {}
+        if self.requires_column_info:
+            return NotImplementedError(
+                f"requires_column_info={self.requires_column_info} not implemented for OVD tasks."
+            )
+
+        fn.update(
+            {
+                self.image_key: PadCollator(pad_val=0),
+                self.prompt_key: ListCollator(),
+                self.image_meta_key: ListCollator(),
+            }
+        )
+        return fn
+
+    def process_one_sample(
+        self,
+        texts_and_images: Dict[str, Union[str, List[str]]],
+        is_training: bool,
+    ) -> Dict:
+        """
+        Read images, process them, and stack them. One sample can have multiple images,
+        resulting in a tensor of (n, 3, size, size), where n <= max_img_num_per_col is the available image number.
+
+        Parameters
+        ----------
+        texts_and_images
+            The input could be image or text (image captions) for open vocabulary detection.
+        is_training
+            Whether to process images in the training mode.
+
+        Returns
+        -------
+        A dictionary containing one sample's images and their number.
+        """
+        ret = {}
+        image_data = {}
+
+        for per_col_name, per_col_content in texts_and_images.items():
+            if is_image_input(per_col_content, IMAGE_PATH):
```

Comment:
We can use the `feature_modalities ` to check whether the modality type is image_path. Refer to `process_image.py`. `is_image_input ` is not necessary.

Reply:
Done.